components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -14,89 +14,527 @@
 
 #import <XCTest/XCTest.h>
 
+#import <objc/runtime.h>
 #import "MaterialTextFields+ContainedInputView.h"
 
-@interface MDCBaseTextField (Testing)
+@interface MDCBaseTextField (Private)
 @property(nonatomic, assign) UIUserInterfaceLayoutDirection layoutDirection;
+- (CGRect)adjustTextAreaFrame:(CGRect)textRect
+    withParentClassTextAreaFrame:(CGRect)parentClassTextAreaFrame;
+@end
+
+@interface MDCBaseTextField (Testing)
+@property(nonatomic, assign) BOOL isEditingOverride;
+@end
+
+@implementation MDCBaseTextField (Testing)
+@dynamic isEditingOverride;
+- (BOOL)isEditingOverride {
+  return [objc_getAssociatedObject(self, _cmd) boolValue];
+}
+- (void)setIsEditingOverride:(BOOL)isEditingOverride {
+  objc_setAssociatedObject(self, @selector(isEditingOverride), @(isEditingOverride),
+                           OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+- (BOOL)isEditing {
+  return self.isEditingOverride ? self.isEditingOverride : [super isEditing];
+}
```

Comment:
Overriding an existing method in a class can potentially lead to flaky tests because the order in which overloaded category methods are loaded is undefined.

Did you find that becoming the first responder was not sufficient within the test harness? If so then either a private test-only API or a subclass are two alternative solutions here.

Reply:
Yeah this is sketchy, I'm just trying a bunch of random stuff right now to get the tests to pass in kokoro. Having a hard time!